src/zx/FunctionalityConstruction.cpp

```diff
@@ -128,23 +149,60 @@ void FunctionalityConstruction::addRzx(ZXDiagram& diag,
   addXSpider(diag, target2, qubits);
 
   const auto midX =
-      diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)), zx::VertexType::X);
-  const auto midZ = diag.addVertex(-1, -1, phase, zx::VertexType::Z);
+      diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)), VertexType::X);
+  const auto midZ = diag.addVertex(-1, -1, phase, VertexType::Z);
   diag.addEdge(qubits[static_cast<std::size_t>(target)], midX);
   diag.addEdge(qubits[static_cast<std::size_t>(target2)], midX,
                EdgeType::Hadamard);
   diag.addEdge(midX, midZ);
-  diag.addGlobalPhase(-phase / 2.0);
+  diag.addGlobalPhase(-phase / 2);
 }
 
-void FunctionalityConstruction::addDcx(zx::ZXDiagram& diag,
-                                       const zx::Qubit qubit1,
-                                       const zx::Qubit qubit2,
+void FunctionalityConstruction::addDcx(ZXDiagram& diag, const Qubit qubit1,
+                                       const Qubit qubit2,
                                        std::vector<Vertex>& qubits) {
   addCnot(diag, qubit1, qubit2, qubits);
   addCnot(diag, qubit2, qubit1, qubits);
 }
 
+void FunctionalityConstruction::addXXplusYY(
+    ZXDiagram& diag, const PiExpression& theta, const PiExpression& beta,
+    const Qubit qubit0, const Qubit qubit1, std::vector<Vertex>& qubits) {
+  addRz(diag, beta, qubit1, qubits);
+  addRz(diag, PiExpression(PiRational(-1, 2)), qubit0, qubits);
+  addRx(diag, PiExpression(PiRational(1, 2)), qubit0, qubits);
+  addRz(diag, PiExpression(PiRational(1, 2)), qubit0, qubits);
+  addZSpider(diag, qubit1, qubits, PiExpression(PiRational(1, 2)));
```

Comment:
The S and the RZ on qubit1 can be fused by constructing the RZ with an additional phase of pi/2 and removing the `addZSpider`.

Reply:
Done in https://github.com/cda-tum/mqt-core/pull/482/commits/5b061782e6394cfd7e08fc1638b088962848fbd8